src/main/java/hw15/model/FileManager.java

```diff
@@ -0,0 +1,104 @@
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
+            numberOfWords = numberOfWords + songArray.length;
+        }
+        System.out.println("The quantity of all words in the song :" + numberOfWords);
+        bufferedReader.close();
+    }
+
+    public void writeShortWordsToNewFile() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        FileWriter fileWriter = new FileWriter(newFileName);
+        BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            String[] shortWordsArray = line.split(" ");
+
+            for (String s : shortWordsArray) {
+                if (s.length() <= LETTERS_LIMIT && !s.equals("" + "")) {
+                    bufferedWriter.write(s + "\n");
+                    bufferedWriter.flush();
+
+                    numberOfWords++;
+
+                }
+            }
+        }
+        System.out.println("The quantity of short words in the song :" + numberOfWords + "\n");
+        bufferedReader.close();
+        bufferedWriter.close();
+    }
+
+    public void readShortWordsFromNewFile() throws IOException {
+        FileReader fileReader = new FileReader(newFileName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        System.out.println("Read from a new file: ");
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            System.out.print(line + ", ");
+        }
+        System.out.println("\n");
+        bufferedReader.close();
+    }
+
+    public void writeAllWordsToList() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        String line;
+
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
+            int i = 0;
+            for (String s : songArray) {
+                if (!s.equals("" + "")) {
+                    listOfAllWords.add(i, s.toLowerCase());
+                }
+                i++;
+            }
+        }
+        bufferedReader.close();
+    }
+
+    public void showMostCommonWords() {
+        Map<String, Long> mostCommonWords = listOfAllWords.stream()
+                .collect(Collectors.groupingBy(listOfAllWords -> listOfAllWords, Collectors.counting()));
+
+        List<Map.Entry<String, Long>> collect = mostCommonWords.entrySet().stream()
+                .sorted(Map.Entry.<String, Long>comparingByValue().reversed())
+                .collect(Collectors.toList());
```

Comment:
try to make Map with `NUMBER_OF_POPULAR_WORDS` amount
and y will  get rid of for loop lower

Reply:
+